mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxEventDispatcher.java

```diff
@@ -56,23 +56,20 @@
  * Helper class to dispatch {@link org.apache.james.mailbox.Event}'s to registerend MailboxListener
  */
 public class MailboxEventDispatcher {
-
     @VisibleForTesting
     public static MailboxEventDispatcher ofListener(MailboxListener mailboxListener) {
-        return new MailboxEventDispatcher(mailboxListener, new EventFactory());
+        return new MailboxEventDispatcher(mailboxListener);
     }
 
     private final MailboxListener listener;
-    private final EventFactory eventFactory;
 
     @Inject
     public MailboxEventDispatcher(DelegatingMailboxListener delegatingMailboxListener) {
-        this(delegatingMailboxListener, new EventFactory());
+        this((MailboxListener) delegatingMailboxListener);
     }
 
-    private MailboxEventDispatcher(MailboxListener listener, EventFactory eventFactory) {
+    private MailboxEventDispatcher(MailboxListener listener) {
```

Comment:
you probably don't need this constructor anymore

Reply:
We do still need regisiter "MailboxListener" in tests.

IMO we can add the "delegating" layer in the tests